src/coreclr/gc/gc.cpp

```diff
@@ -40349,12 +40349,7 @@ HRESULT GCHeap::Initialize()
         {
             nhp = min(nhp, num_affinitized_processors);
         }
-#ifndef TARGET_WINDOWS
-        // Limit the GC heaps to the number of processors available in the system.
-        nhp = min (nhp, GCToOSInterface::GetTotalProcessorCount());
-#endif // !TARGET_WINDOWS
```

Comment:
This clipping was needed by the previous version of the code.  The number of GC heaps is still limited by `MAX_SUPPORTED_CPUS` (always) and `process_affinity_set->Count()` (if affinitizing).